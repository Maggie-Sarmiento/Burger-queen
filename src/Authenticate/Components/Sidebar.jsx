/* eslint-disable react/prop-types */
/* eslint-disable arrow-body-style */
import React from 'react';
import styles from './Sidebar.module.css';

const Sidebar = ({ rol }) => {
  if (rol === 'Admin') {
    return (
      <section className={styles.sidebar}>
        <img className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <button type="button" className={`${styles.btn} ${styles.btnStaff}`}>Empleados</button>
        <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Menus</button>
      </section>
    );
  }

  if (rol === 'Mesero') {
    return (
      <section className={styles.sidebar}>
        <img className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <button type="button" className={`${styles.btn} ${styles.btnStaff}`}>Menus</button>
        <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Pedidos</button>
        <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Servir</button>
      </section>
    );
  }

  if (rol === 'Jefe de cocina') {
    return (
      <section className={styles.sidebar}>
        <img className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <button type="button" className={`${styles.btn} ${styles.btnStaff}`}>Pedidos</button>
        <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Servir</button>
      </section>
    );
  }

  return null;
};

export default Sidebar;
