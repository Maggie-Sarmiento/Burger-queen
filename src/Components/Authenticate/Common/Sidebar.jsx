/* eslint-disable react/prop-types */
/* eslint-disable arrow-body-style */
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './Sidebar.module.css';

const Sidebar = ({ role }) => {
  if (role === 'Admin') {
    return (
      <section className={styles.sidebar}>
        <img className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <Link to="staff">
          <button type="button" className={`${styles.btn} ${styles.btnStaff}`}>Empleados</button>
        </Link>
        <Link to="menu">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Menu Desayuno</button>
        </Link>
        <Link to="menu-lunch">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Menu Vespertino</button>
        </Link>
      </section>
    );
  }

  if (role === 'Mesero') {
    return (
      <section className={styles.sidebar}>
        <img className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <Link to="menu">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Menu Desayuno</button>
        </Link>
        <Link to="menuLunch">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Menu Almuerzo</button>
        </Link>
        <Link to="order">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Pedidos</button>
        </Link>
        <Link to="toserve">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Servir</button>
        </Link>
        <Link to="list-order">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Lista Pedidos</button>
        </Link>
      </section>
    );
  }

  if (role === 'Jefe de cocina') {
    return (
      <section className={styles.sidebar}>
        <img className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <Link to="order">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Pedidos</button>
        </Link>
        <Link to="toserve">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Servir</button>
        </Link>
      </section>
    );
  }

  return null;
};

export default Sidebar;
