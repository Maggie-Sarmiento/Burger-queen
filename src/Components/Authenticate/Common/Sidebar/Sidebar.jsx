/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable react/prop-types */
/* eslint-disable arrow-body-style */
import React from 'react';
import { Link, useNavigate } from 'react-router-dom';
import styles from './Sidebar.module.css';

const Sidebar = ({ role, onClick }) => {
  const navigate = useNavigate();
  const goToHome = () => {
    navigate('/home');
    onClick();
  };

  if (role === 'Admin') {
    return (
      <section className={styles.sidebar}>
        <img onClick={goToHome} className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <Link to="staff">
          <button type="button" className={`${styles.btn}`}>Empleados</button>
        </Link>
        <hr />
        <Link to="menu">
          <button type="button" className={`${styles.btn}`}>Menu Desayuno</button>
        </Link>
        <hr />
        <Link to="menu-lunch">
          <button type="button" className={`${styles.btn}`}>Menu Vespertino</button>
        </Link>
      </section>
    );
  }

  if (role === 'Mesero') {
    return (
      <section className={styles.sidebar}>
        <img onClick={goToHome} className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <Link to="menu">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Menu Desayuno</button>
        </Link>
        <hr />
        <Link to="menu-lunch">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Menu Almuerzo</button>
        </Link>
        <hr />
        <Link to="order">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Pedidos</button>
        </Link>
        <hr />
        <Link to="toserve">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Servir</button>
        </Link>
        <hr />
        <Link to="list-orders">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Lista Pedidos</button>
        </Link>
      </section>
    );
  }

  if (role === 'Jefe de cocina') {
    return (
      <section className={styles.sidebar}>
        <img onClick={goToHome} className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
        <Link to="order">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Pedidos</button>
        </Link>
        <hr />
        <Link to="toserve">
          <button type="button" className={`${styles.btn} ${styles.btnMenus}`}>Servir</button>
        </Link>
      </section>
    );
  }

  return null;
};

export default Sidebar;
