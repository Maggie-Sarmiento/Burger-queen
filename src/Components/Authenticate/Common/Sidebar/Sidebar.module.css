.sidebar {
  background-color: #F3B240;
  display: flex;
  flex-direction: column;
  height: 100vh;
  align-items: center;
}

.logoSidebar {
  width:90%;
  margin-bottom: 3em;
  margin-top: 1em;
}

hr {
  border: 0;
  height: 1px;
  background-color: black;
  width: 80%;
  margin-bottom: 1em;
  margin-bottom: 1em;
}

.btn {
  background-color: #F3B240;
  border: none;
  cursor: pointer;
  font-size: 0.6em;
  height: 2.7em;
  margin-bottom: 3em;
  width: auto;
  transform: rotate(-90deg);
}

.btn:hover {
  background-color: rgba(255, 255, 255, 0.6);
  border-radius: 0.8em;
  height: 2.7em;
}
.btn:focus {
  background-color: rgba(255, 255, 255, 0.6);
  border-radius: 0.8em;
  height: 2.7em;
}

@media (min-width: 450px) {
  .btn{
    margin-top: 2em;
  }
}

@media (min-width: 540px) {
  .btn {
    transform: rotate(0);
    margin-bottom: 1em;
    margin-top: 1em;
    font-size: 0.6em;
    height: 2.7em;
  }

  .logoSidebar {
    margin-bottom: 2em;
  }

  hr {
    border: 0;
    height: 0;
    width: 0;
    margin: 0;
  }
}

@media (min-width: 850px) {
  .btn {
    font-size: 0.6em;
    height: 2.7em;
  }
}

@media (min-width: 1000px) {
  .btn {
    font-size: 0.8em;
    height: 2.7em;
  }
}