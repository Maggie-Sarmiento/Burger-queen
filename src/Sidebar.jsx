import React from 'react';
import styles from './Sidebar.module.css'

const Sidebar = () => {
  return (
    <section className={styles.sidebar}>
      <img className={styles.logoSidebar} src="https://svgshare.com/i/gmq.svg" alt="logo" />
      <button className={`${styles.btn} ${styles.btnStaff}`}>Empleados</button>
      <button className={`${styles.btn} ${styles.btnMenus}`}>Menus</button>
    </section>
  );
}

export default Sidebar;
