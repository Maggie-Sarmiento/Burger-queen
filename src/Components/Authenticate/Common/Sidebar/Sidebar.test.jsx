/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import Sidebar from './Sidebar';

describe('test Sidebar', () => {
  test('render component Sidebar', async () => {
    let root;
    const mockAuthenticate = jest.fn();
    act(() => {
      root = create(
        <Router>
          <Sidebar authenticate={mockAuthenticate} />
        </Router>,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
