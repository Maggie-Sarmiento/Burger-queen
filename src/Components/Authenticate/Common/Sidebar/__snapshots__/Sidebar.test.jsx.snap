// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Sidebar render component Sidebar 1`] = `null`;
